packages/expect-utils/src/jasmineUtils.ts

```diff
@@ -112,10 +112,9 @@ function eq(
         return Object.is(a.valueOf(), b.valueOf());
       }
     case '[object Date]':
-      // Coerce dates to numeric primitive values. Dates are compared by their
-      // millisecond representations. Note that invalid dates with millisecond representations
-      // of `NaN` are not equivalent.
-      return +a == +b;
+      // Coerce dates to numeric primitive values.
+      // Dates are compared by their millisecond representations.
+      return Object.is(+a, +b);
```

Comment:
Just curious. Did you try benchmarking something like this (if it does not break tests, of course):

```suggestion
      return Object.is(Number(a), Number(b));
```

Reply:
wrapped by Number contructor case is slower than +(plus) case